Adyen/adyen-magento2
# Recurring payments added functionality

**Description**
This PR will add the following functionality:

* CardOnFile / Subscription / UnscheduledCardOnFile payments trough GooglePay
* Subscription / UnscheduledCardOnFile trough PayPal
* CardonFile / UnscheduledCardOnFile trough Magento Vault card payments
* Customer token removal

**Tested scenarios**
1. Set vault token type to **CardOnFile**, on both the card and the payment method configuration and test the following:
    * Initial Google Pay payment
    * Tokenised Google Pay payment
    * Card payment
    * Tokenised card payment
    * Delete tokens from customer page

2. Set vault token type to **Subscription**, on both the card and the payment method configuration and test the following:
    * Initial Google Pay payment
    * Check token in `vault_payment_token` created with `tokenType = Subscription`
    * Initial card payment
    * Check token in `vault_payment_token` created with `tokenType = Subscription`
    * Initial Paypal payment
    * Check token in `vault_payment_token` created with `tokenType = Subscription`

3. Set vault token type to **UnscheduledCardOnFile**, on both the card and the payment method configuration and test the following:
    * Initial Google Pay payment
    * Check token in `vault_payment_token` created with `tokenType = UnscheduledCardOnFile`
    * Initial card payment
    * Check token in `vault_payment_token` created with `tokenType = UnscheduledCardOnFile`
    * Initial Paypal payment
    * Check token in `vault_payment_token` created with `tokenType = UnscheduledCardOnFile`
